lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -4167,9 +4173,16 @@ namespace {
       }
 
       case ActorIsolation::ActorInstance: {
-        if (auto param = closure->getCaptureInfo().getIsolatedParamCapture())
-          return ActorIsolation::forActorInstanceCapture(param)
-            .withPreconcurrency(preconcurrency);
+        if (checkIsolatedCapture) {
+          if (auto param = closure->getCaptureInfo().getIsolatedParamCapture())
+            return ActorIsolation::forActorInstanceCapture(param)
+                .withPreconcurrency(preconcurrency);
+        } else {
+          // If we don't have capture information during code completion, assume
+          // that the closure captures the `isolated` parameter from the parent
+          // context.
+          return parentIsolation;
+        }
```

Comment:
I guess an alternative solution would be to call `computeCaptures` for completion to just populate an empty set of captures. Can't say I really love either solution, but I think I would mildly prefer `computeCaptures` since it's less invasive.